### PR TITLE
chores(claims): remove `x/claims` module

### PR DESCRIPTION
**Pull request by ronal999999**
Read more at #41

Closes: #41

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -79,6 +79,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 - (revenue) [#68](https://github.com/EscanBE/evermint/pull/68) Remove `x/revenue` module
 - (recovery) [#69](https://github.com/EscanBE/evermint/pull/69) Remove `x/recovery` module
 - (epochs) [#70](https://github.com/EscanBE/evermint/pull/70) Remove `x/epochs` module
+- (claim) [#71](https://github.com/EscanBE/evermint/pull/73) Remove `x/claim` module
```
Comment (VictorEscan) [inline CHANGELOG.md:82]:
```suggestion
- (claims) [#73](https://github.com/EscanBE/evermint/pull/73) Remove `x/claims` module
```

File: app/app.go [hunk 2]
```diff
@@ -479,43 +467,39 @@ func NewEvermint(
...
 		evmkeeper.NewMultiEvmHooks(
 			chainApp.Erc20Keeper.Hooks(),
-			chainApp.ClaimsKeeper.Hooks(),
 		),
 	)
 
 	chainApp.TransferKeeper = transferkeeper.NewKeeper(
 		appCodec, keys[ibctransfertypes.StoreKey], chainApp.GetSubspace(ibctransfertypes.ModuleName),
-		chainApp.ClaimsKeeper, // ICS4 Wrapper: claims IBC middleware
+		chainApp.IBCKeeper.ChannelKeeper, // ICS4 Wrapper: claims IBC middleware
```
Comment (VictorEscan) [inline app/app.go:487]:
```suggestion
		chainApp.IBCKeeper.ChannelKeeper, // No ICS4 wrapper
```

File: ibc/utils_test.go [hunk 3]
```diff
@@ -203,20 +202,20 @@ func TestGetTransferAmount(t *testing.T) {
...
+		//			&transfertypes.FungibleTokenPacketData{
+		//				Sender:   "cosmos1qql8ag4cluz6r4dz28p3w00dnc9w8ueulg2gmc",
+		//				Receiver: marker.ReplaceAbleAddress("evm1x2w87cvt5mqjncav4lxy8yfreynn273xe08fl7"),
+		//				Amount:   types.IBCTriggerAmt,
+		//			},
+		//		),
+		//	},
+		//	types.IBCTriggerAmt,
+		//	false,
+		//},
```
Comment (VictorEscan) [inline ibc/utils_test.go:218]: Why delete test?

File: x/erc20/keeper/ibc_callbacks_integration_test.go [hunk 4]
```diff
@@ -214,72 +212,72 @@ var _ = Describe("Convert receiving IBC to Erc20", Ordered, func() {
...
+	//		// validate that Osmosis address balance is correct
+	//		uosmoFinalBalance := s.IBCOsmosisChain.GetSimApp().BankKeeper.GetBalance(s.IBCOsmosisChain.GetContext(), senderAcc, "uosmo")
+	//		s.Require().Equal(uosmoInitialBalance.Amount.Int64()-amount, uosmoFinalBalance.Amount.Int64())
+	//
+	//		// validate that Receiver address on Evermint got the claims tokens
+	//		receiverFinalNativeCoinBalance := s.app.BankKeeper.GetBalance(s.EvermintChain.GetContext(), receiverAcc, constants.BaseDenom)
+	//
+	//		s.Require().Equal(receiverInitialNativeCoinBalance.Amount.Add(claimableAmount).Sub(sendBackCoinsFee), receiverFinalNativeCoinBalance.Amount)
+	//	})
+	//})
```
Comment (VictorEscan) [inline x/erc20/keeper/ibc_callbacks_integration_test.go:280]:
```suggestion
```
Delete this test

File: app/app.go [hunk 5]
```diff
@@ -684,7 +662,6 @@ func NewEvermint(
 		banktypes.ModuleName,
 		distrtypes.ModuleName,
 		// NOTE: staking requires the claiming hook
```
Comment (VictorEscan) [inline app/app.go:664]:
```suggestion
```
Delete this comment

File: x/erc20/keeper/ibc_callbacks_test.go [hunk 6]
```diff
@@ -627,7 +621,7 @@ func (suite *KeeperTestSuite) TestOnTimeoutPacket() {
 			name: "no-op - sender is module account",
 			malleate: func() transfertypes.FungibleTokenPacketData {
 				// any module account can be passed here
-				moduleAcc := suite.app.AccountKeeper.GetModuleAccount(suite.ctx, "claims")
+				moduleAcc := suite.app.AccountKeeper.GetModuleAccount(suite.ctx, types.ModuleName)
```
Comment (VictorEscan) [inline x/erc20/keeper/ibc_callbacks_test.go:624]: TODO check logic

File: x/erc20/keeper/ibc_callbacks_test.go [hunk 7]
```diff
@@ -246,12 +245,8 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
 			expCoins:      coins,
 		},
 		{
-			name: "ibc conversion - sender == receiver and from evm chain",
+			name: "ibc conversion - sender == receiver",
```
Comment (VictorEscan) [inline x/erc20/keeper/ibc_callbacks_test.go:248]: TODO check logic

File: x/erc20/keeper/ibc_callbacks.go [hunk 8]
```diff
@@ -56,14 +56,6 @@ func (k Keeper) OnRecvPacket(
 		return channeltypes.NewErrorAcknowledgement(err)
 	}
 
-	claimsParams := k.claimsKeeper.GetParams(ctx)
```
Comment (VictorEscan) [inline x/erc20/keeper/ibc_callbacks.go:59]: TODO: check logic & behavior

File: x/erc20/keeper/utils_test.go [hunk 9]
```diff
@@ -209,10 +208,6 @@ func (suite *KeeperTestSuite) SetupIBCTest() {
 	err = suite.IBCCosmosChain.GetSimApp().BankKeeper.SendCoinsFromModuleToAccount(suite.IBCCosmosChain.GetContext(), minttypes.ModuleName, suite.IBCCosmosChain.SenderAccount.GetAddress(), stkCoin)
 	suite.Require().NoError(err)
 
-	claimparams := claimstypes.DefaultParams()
-	claimparams.AirdropStartTime = suite.EvermintChain.GetContext().BlockTime()
-	claimparams.EnableClaims = true
-	err = s.app.ClaimsKeeper.SetParams(suite.EvermintChain.GetContext(), claimparams)
 	suite.Require().NoError(err)
```
Comment (VictorEscan) [inline x/erc20/keeper/utils_test.go:211]: Delete this line

**Review by VictorEscan — CHANGES_REQUESTED** [review]

File: ibc/utils_test.go (hunk 3 above)
Comment (VictorEscan) [inline ibc/utils_test.go:218]:
```suggestion
		{
			"valid - any amount",
			channeltypes.Packet{
				Data: transfertypes.ModuleCdc.MustMarshalJSON(
					&transfertypes.FungibleTokenPacketData{
						Sender:   "cosmos1qql8ag4cluz6r4dz28p3w00dnc9w8ueulg2gmc",
						Receiver: marker.ReplaceAbleAddress("evm1x2w87cvt5mqjncav4lxy8yfreynn273xe08fl7"),
						Amount:   "1",
					},
				),
			},
			"1",
			false,
		},
```

**Review by VictorEscan — CHANGES_REQUESTED** [review]

File: x/erc20/keeper/ibc_callbacks_test.go (hunk 7 above)
Comment (VictorTrustyDev) [reply x/erc20/keeper/ibc_callbacks_test.go:248]: Ref #79

File: x/erc20/keeper/ibc_callbacks.go (hunk 8 above)
Comment (VictorTrustyDev) [reply x/erc20/keeper/ibc_callbacks.go:59]: Ref #79

**Review by VictorEscan — APPROVED** [review]

**Pull request merged**